src/components/ShoppingCart/ShoppingCart.component.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import PropTypes from 'prop-types';
+
+import ShoppingCartTable from './ShoppingCartTable/ShoppingCartTable.styles';
+import { useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+
+import { checkoutActionCreators } from 'state/index';
+import { bindActionCreators } from 'redux';
+import { useDispatch } from 'react-redux';
+
+const ShoppingCartComponent = ({className}) => {
+    const { items } = useSelector((state) => state.shoppingCart);
+    const dispatch = useDispatch()
+    const { setCheckoutSummary } = bindActionCreators(checkoutActionCreators, dispatch);
+    const history = useHistory();
+
+    const calculateTotalPrice = (items) => {
+        const totalPrice = items.reduce( (previous, current) => previous + (current.price*current.quantity),0);
+        return totalPrice;
```

Comment:
For this i think i would have used an useEffect 
```suggestion
    const { totalPrice, setTotalPrice } = useState(0);
    useEffect(() => {
       setTotalPrice(items.reduce( (previous, current) => previous + (current.price*current.quantity),0));
    }, [items]);
```
on line `45` 
```
<div className="shoppingCart-total">Total: ${ totalPrice }</div>
```
and on `33`
```
const checkoutData = { orderSummary, total: totalPrice}
```